test/expectation.jl

```diff
@@ -96,3 +96,31 @@ end;
         end
     end
 end
+
+@testset "Variance reduction" begin
+    for seed in 1:10
+        rng = StableRNG(seed)
+        f(x) = x
+        dist_constructor(θ) = MvNormal(θ, I)
+        n = 10
+        θ = randn(rng, n)
+        r = Reinforce(
+            f, dist_constructor; rng=rng, nb_samples=100, seed=seed, variance_reduction=true
+        )
+        r_no_variance_reduction = Reinforce(
+            f,
+            dist_constructor;
+            rng=rng,
+            nb_samples=100,
+            seed=seed,
+            variance_reduction=false,
+        )
+
+        J_reduced_variance = jacobian(r, θ)[1]
+        J_no_reduced_variance = jacobian(r_no_variance_reduction, θ)[1]
+        J_true = Matrix(I, n, n)
+
+        mape(x::AbstractArray, y::AbstractArray) = mean(abs.(x .- y))
+        @test mape(J_reduced_variance, J_true) < mape(J_no_reduced_variance, J_true)
```

Comment:
Does this really test what we want? I would rather test that the actual variance decreases. It's not unlikely to get a result without variance reduction that happens to be very close to the ground truth.

Reply:
Pushed new tests checking directly that the variance is lower. What do you think?

Reply:
I modified them a little bit, can you review that everything is okay for you and then we merge?